core/jobs/types/base_validation_errors.py

```diff
@@ -49,11 +64,24 @@ def __init__(self, message, model_or_kind, model_id=None):
             raise ValueError('message must be a non-empty string')
 
         if model_id is None:
+            # Here, get_model_id and get_model_kind can only accept argument
+            # of type base_models.BaseModel but model_or_kind is defined as a
+            # type of Union[base_models.BaseModel, str]. So, to rule out the
+            # possibility of str of model_or_kind for mypy type checking.
+            # We asserted it as an Basemodel type.
+            assert isinstance(model_or_kind, base_models.BaseModel)
```

Comment:
Maybe we can add this instance check into the `if`?

Reply:
Done!